### ISSUE TITLE
Performance of where method

### ISSUE THREAD

**Issue by czeslaaw** [Exygy/minimongoid#26]
```
  @where: (selector = {}, options = {}) ->
    result = @modelize @find(selector, options)
    result.setQuery selector
    result
```

I noticed that where() is invalidated on every change in cursor returned by find. Is it possible to do this by using transform option on find? That way we would only do fetch and init when we need to.

Something like this:
```
  @where: (selector = {}, options = {}) ->
    options = _.extend({transform (doc) -> @init(doc)}, options)
    @find(selector, options)
```

**Comment by kaptron** [comment]
@czeslaaw you bring up a good point; however in practice (for me at least) the `where()` method is intended to be a "fetched" representation of the docs in question. Meaning, when I call `where()` I'm expecting to get an array of documents back that can be manipulated/treated as such (using underscore methods for sorting, etc.). That was somewhat the purpose of the difference between the `where()` and `find()` methods; `find()` returns a cursor rather than an Array. 

But you pointed out a flaw in `find()` which is that it was just giving you a straightforward `Collection.find()`, which meant that you would just be getting plain documents back, and not models. e.g. `MyModel.find({property: 'xyz'})` wasn't going to return instances of MyModel unless you explicitly called `MyModel.init()` on the results. 

Check out dec87ab060ee1736f87770a819e9445dc182f800 (and update your package to v0.9.4), and you'll see what I did was move the "modelizing" logic to instead be a "transform" inside of `find()`. So *now* if you use the `find()` method you will automatically get transformed models upon fetching the results. And the `where()` method differs by automatically calling `fetch()` which gives you back an Array (technically, a Relation) of results. 

In summary -- use `Model.find()` when you're just wanting a cursor. A good example of this is when setting up your publish methods in your server you can return `Model.find(selector)`. Another example where I'd only need the cursor is in Iron Router, when passing data to the template. But when I'm purposefully wanting to fetch the results in an array, I use `Model.where(selector)`.

Thinking about it further, I suppose the `where()` method might be more properly named (or aliased) as `fetch()`.

**Comment by czeslaaw** [comment]
That's what i needed. Thanks!
